components/src/main/java/org/folio/rest/workflow/utility/CompressFileMagic.java

```diff
@@ -0,0 +1,144 @@
+package org.folio.rest.workflow.utility;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.folio.rest.workflow.enums.CompressFileFormat;
+
+/**
+ * Provide functionality to detect file mime-types based on the magic headers.
+ *
+ * This approach is more reliable than detecting via file-extensions.
+ *
+ * @see <a href="https://en.wikipedia.org/wiki/List_of_file_signatures">(Wikipedia) List of Magic Headers</a>
+ */
+public class CompressFileMagic {
+
+  private CompressFileMagic() {
+    throw new IllegalStateException("Attempt to instatiate a utility class.");
```

Comment:
Interesting. A bit excessive but will prevent a developer from calling private constructor in the this utility class.

Reply:
This is required by the SonarCloud as it considers **not** doing this a Code Smell.

Reply:
I have seen the make a private constructor recommendation.  But classifying missing dead code as a code smell is somewhat absurd. I will update my definition of code smell with put in place safety rails for developers to not unintentionally call a private constructor in a utility class.